pandas/core/arrays/datetimelike.py

```diff
@@ -402,12 +402,8 @@ def __array__(self, dtype=None):
         return self._data
 
     @property
-    def size(self) -> int:
-        """The number of elements in this array."""
-        return np.prod(self.shape)
-
-    def __len__(self):
-        return len(self._data)
+    def shape(self):
```

Comment:
you should remove and len definitions unless they are needed

Reply:
is there a missing word in this comment?  This already removes all the relevant redundant methods.